cypress/integration/tests/administrator/custom-migration-targets/crud.test.ts

```diff
@@ -15,46 +15,125 @@ limitations under the License.
-        if (hasToBeSkipped("@tier1") && hasToBeSkipped("@dc")) return;
-
         login();
 
+        cy.fixture("custom-migration-targets").then(function (customMigrationTargets) {
+            this.customMigrationTargets = customMigrationTargets;
+        });
+
         cy.intercept("POST", "/hub/rulebundles*").as("postRule");
         cy.intercept("GET", "/hub/rulebundles*").as("getRule");
         cy.intercept("PUT", "/hub/rulebundles*/*").as("putRule");
         cy.intercept("DELETE", "/hub/rulebundles*/*").as("deleteRule");
     });
 
-    it("Custom Migration Targets CRUD", { tags: ["@tier1", "@dc"] }, function () {
+    it("Custom Migration Targets CRUD with rules uploaded manually", function () {
         CustomMigrationTarget.open();
+
+        const targetData = this.customMigrationTargets.manual_rules;
         const target = new CustomMigrationTarget(
             data.getRandomWord(8),
             data.getDescription(),
-            "img/cloud.png",
-            "xml/javax-package-custom-target.windup.xml"
+            targetData.image,
+            {
+                type: CustomRuleType.Manual,
+                rulesetPaths: targetData.rulesFiles,
+            }
         );
         target.create();
         cy.wait("@postRule");
         cy.contains(CustomMigrationTargetView.takeMeThereNotification).click();
         cy.get("article", { timeout: 12 * SEC }).should("contain", target.name);
 
         const newName = data.getRandomWord(8);
+        const newRules: RulesManualFields = {
+            type: CustomRuleType.Manual,
+            rulesetPaths: ["xml/javax-package-custom.windup.xml"],
+        };
+
         target.edit({
             name: newName,
-            rulesetPath: "xml/javax-package-custom.windup.xml",
+            ruleType: newRules,
         });
         cy.wait("@putRule");
         cy.get("article", { timeout: 12 * SEC }).should("contain", newName);
         target.name = newName;
-        target.rulesetPath = "xml/javax-package-custom.windup.xml";
+        target.ruleType = newRules;
+
+        target.delete();
+        cy.wait("@deleteRule");
+        cy.get("article", { timeout: 12 * SEC }).should("not.contain", target.name);
+    });
+
+    it("Create Custom Migration Target with rules from repository with credentials", function () {
+        const sourceCredential = new CredentialsSourceControlUsername(
+            data.getRandomCredentialsData(
+                CredentialType.sourceControl,
+                UserCredentials.usernamePassword,
+                Cypress.env("git_password") && Cypress.env("git_user")
+            )
+        );
+
+        sourceCredential.create();
+        const targetData = this.customMigrationTargets.rules_from_tackle_testApp;
+        const repositoryData: RulesRepositoryFields = {
+            ...targetData.repository,
+            type: CustomRuleType.Repository,
+            credentials: sourceCredential,
+        };
+
+        CustomMigrationTarget.open();
```

Comment:
Why do we have to open the menu ?
This should be taken care by target.create()

Reply:
You're right, solved.